addons/pos_viva_com/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -16,10 +20,46 @@ patch(PaymentScreen.prototype, {
                 return;
             }
         });
+
+        this.vivaApp = useVivaApp(this.validateOrder.bind(this));
+        onWillStart(async () => {
+            await this.vivaApp.process();
+        });
     },
+    async addNewPaymentLine(pm) {
+        if (this.vivaApp.use(pm)) {
+            const previousAnswer = window.localStorage.getItem("vivawallet_app_answer");
+            if (previousAnswer === null) {
+                await makeAwaitable(this.dialog, ConfirmationDialog, {
+                    title: _t("Viva Wallet Application"),
+                    body: _t(
+                        "Is the vivawallet application installed on your device? If so, the payment will initialize directly from it."
+                    ),
+                    cancelLabel: _t("No"),
+                    confirmLabel: _t("Yes"),
+                    confirm: () => {
+                        window.localStorage.setItem("vivawallet_app_answer", "true");
+                    },
+                    cancel: () => {
+                        window.localStorage.setItem("vivawallet_app_answer", "false");
+                    },
+                });
+
+                return;
+            }
```

Comment:
We asks for `vivawallet_app_answer`, what do we do when the user confirms? The value doesn't seem to be taken into account in the logic.

Reply:
That's true will I change that